src/ShadowDOM/querySelector.js

```diff
@@ -136,6 +136,7 @@
     var target = unsafeUnwrap(this);
     var list;
     var root = getTreeScope(this).root;
+    root = (!deep && root.shadowRoot) || root;
```

Comment:
I would suggest to merge this line into the line above. Would not be much harder to read, but saves calling getTreeScope in case the condition is true.